### PR TITLE
Commerce 2036 Catalog enpoints now support Filter and Sort

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Comment by ethanbustad** [comment]
ci:test
tests should be fixed in upstream now

File: headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/odata/entity/v1_0/SpecificationEntityModel.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * @author Andrea Sbarra
+ */
+public class SpecificationEntityModel implements EntityModel {
+
+	public SpecificationEntityModel() {
+		_entityFieldsMap = Stream.of(
+			new StringEntityField("key", locale -> "key"),
+			new StringEntityField(
+				"title",
+				locale -> "title_".concat(LocaleUtil.toLanguageId(locale)))
```
Comment (ethanbustad) [inline headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/odata/entity/v1_0/SpecificationEntityModel.java]: Should probably use `LocalizationUtil.getLocalizedName` for this

**Review by ethanbustad — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci.test

**Comment by andrea-ale-sbarra** [comment]
Hi @ethanbustad , I've applied your change request, but after that I have this error message in the logs:
`2019-10-28 10:09:30.792 ERROR [fileinstall-/Users/andrea/Documents/work/src/bundles/osgi/modules][LogService:93] Error while starting bundle: file:/Users/andrea/Documents/work/src/bundles/osgi/modules/com.liferay.headless.commerce.admin.catalog.impl.jar
org.osgi.framework.BundleException: Could not resolve module: com.liferay.headless.commerce.admin.catalog.impl [932]_  Unresolved requirement: Import-Package: com.liferay.headless.commerce.admin.catalog.resource.v1_0; version="[3.0.0,4.0.0)"_ [Sanitized]
	at org.eclipse.osgi.container.Module.start(Module.java:444)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.start(EquinoxBundle.java:428)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundle(DirectoryWatcher.java:1263)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundles(DirectoryWatcher.java:1236)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startAllBundles(DirectoryWatcher.java:1225)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.doProcess(DirectoryWatcher.java:515)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.process(DirectoryWatcher.java:365)
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.run(DirectoryWatcher.java:316)
2019-10-28 10:09:30.793 INFO  [fileinstall-/Users/andrea/Documents/work/src/bundles/osgi/modules][BundleStartStopLogger:39] STARTED com.liferay.headless.commerce.admin.catalog.api_5.0.0 [933]
2019-10-28 10:10:21.366 INFO  [fileinstall-/Users/andrea/Documents/work/src/bundles/osgi/modules][BundleStartStopLogger:39] STARTED com.liferay.headless.commerce.admin.catalog.impl_1.0.12 [934]
`
and I can't figure it out what is the cause. Any ideas? Anyway the change seems to be working though.

Thanks!

**Comment by ethanbustad** [comment]
@andrea-ale-sbarra from the error message, it actually looks like it's not a problem to me. It looks like what happened was:

1) `com.liferay.headless.commerce.admin.catalog.impl` deployed, but `com.liferay.headless.commerce.admin.catalog.api` hadn't been deployed yet, so the Impl module failed to register
2) `com.liferay.headless.commerce.admin.catalog.api` deployed and registered
3) `com.liferay.headless.commerce.admin.catalog.impl` registered now that the API module was registered

So I think everything is ok. Let me know if you still have unresolved questions on this.

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Comment by andrea-ale-sbarra** [comment]
this PR now solves also COMMERCE-2215

**Comment by ethanbustad** [comment]
ci:test

**Review by ethanbustad — APPROVED** [review]

**Pull request merged**